### DISCUSSION TITLE
Add options to disable display entities ticking

### DISCUSSION THREAD

**Discussion by galacticwarrior9** [PaperMC/Paper#10533] [Ideas]
### Is your feature request related to a problem?

Display entities tick, but this is unnecessary unless you need them as passengers.

### Describe the solution you'd like.

Separate options to disable ticking all four display entities (Display, TextDisplay, BlockDisplay and ItemDisplay) - and, if possible, interaction entities as well. 

### Describe alternatives you've considered.

I don't believe there are any. 

### Other

Someone proposed this on the Paper Discord last March, but I don't believe they followed through with making an issue.

![Image](https://i.gyazo.com/e2cb99098beb557a10d24d94af1b0584.png)
![Image 2](https://i.gyazo.com/8ecdc2deee53fdb86f0d49a46f7f41b2.png)

**Comment by kennytv** [comment]
I can't really think of a scenario where you would have enough display entities for this to become relevant where you also don't require their passenger/vehicle ticking functionality

**Comment by galacticwarrior9** [comment]
Custom crops, furniture, etc are our current use-cases. The former in particular leads to thousands of DEs being loaded. Our use-case is such that we are not using the display entities as passengers, nor as having passengers.

**Comment by MiniDigger** [comment]
do you happen to have a profiler report?

**Comment by Brokkonaut** [comment]
In contrast to armor stands, which can be affected by collisions, gravity or flowing water, display entities do practically nothing in their tick method. Therefore, it really should not be necessary to prevent them from ticking.

**Comment by Darwin1546** [comment]
I had a 40k display entities in the world and tps drops to 10 just because they exist.  
Disabling ticking is most needed feature

**Comment by masmc05** [comment]
> I had a 40k display entities in the world and tps drops to 10 just because they exist. Disabling ticking is most needed feature

Don't forget that entities have other lags than just ticking, for example they need always to calculate who can see them, and if they need to update their data to players

Even if paper fully disables display entities ticking, 40k will still have a big impact on the server

**Comment by MiniDigger** [comment]
we need a proper profiler report.

**Comment by Malfrador** [comment]
[Spark report with 50k item displays](https://spark.lucko.me/6OB3nieCst)

As already suspected by some, the main issue (nearly 60% of tick time) seems to be the canSee checks and sending the updates to players, not the actual tick logic of the item display itself.
I am not against adding such an option, seems easy enough. But I don't think it will have the positive effects you are hoping for.

**Comment by Darwin1546** [comment]
Yeah, that too  
Looks like minecraft wasn't ready for display entities  
Is there is any fixes for canSee lags problem?

**Comment by Malfrador** [comment]
canSee is barely a real issue, I mixed that up a bit. The main performance loss comes just from tracking that many entities. 
You can improve that easily with a lower entity tracking range (spigot.yml). 50k entities at once also heavily slow the client down.
I really wouldn't blame the game there though - it's just way way more than it was ever designed for.

**Comment by JustAHuman-xD** [comment]
Just like the last person said, I've worked with many display entities and have created various APIs for using and creating complex models with them. The lag that comes from display entities has nothing to do with them ticking and everything to do with tracking them, the more you have the laggier any getEntity, getNearbyEntities, etc methods will be and there isn't really a solution to that unless we made it skip them or something.

I'm all for trying to find ways to reduce their impact but ticking isn't the issue

**Comment by JustAHuman-xD** [comment]
If it'd be useful I can pull up some old reports and stats me and a few others have gotten on this (if the reports haven't expired anyways)

**Comment by Darwin1546** [comment]
Paper should do something with entity handler  
Static display entities with no ai should not kill the server

For example, there is a plugin called FartherViewDistance. It can send chunks to players without loading any entities.  
It's life saver for now, but it's not enough. Even 15k display entities already drops tps to 16
